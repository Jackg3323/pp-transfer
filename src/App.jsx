import { useState } from "react";
import "./App.css";
import Form from "./components/Form/Form";

function App() {
  const [balance, setBalance] = useState(100);
  const [outputMsg, setMsg] = useState("");

  const handleFocus = (event) => {
    console.log("focused");
  };
  const handleBlur = (event) => {
    console.log("Blurred");
  };

  return (
    <Form
      balance={balance}
      setBalance={setBalance}
      setMsg={setMsg}
      outputMsg={outputMsg}
    />
  );
}

export default App;
