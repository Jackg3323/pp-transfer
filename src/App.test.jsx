import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import App from "./App";

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("attempts to transfer an amount from the balance", () => {
  const transferAmount = 10;

  render(<App />);

  const input = screen.getByLabelText(/transfer/i);
  const transferBtn = screen.getByRole("button", { name: "transfer" });

  userEvent.type(input, transferAmount);
  userEvent.click(transferBtn);

  const h2 = screen.getByRole("heading", { level: 2 });

  expect(h2.textContent).toContain(100 - transferAmount);
});
