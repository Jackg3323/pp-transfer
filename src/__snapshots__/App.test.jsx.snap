// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <form
    className="mx-auto w-80 flex-col items-center"
    onSubmit={[Function]}
  >
    <div>
      <label
        className="sr-only"
        htmlFor="transfer"
      >
        Transfer
      </label>
      <input
        className="rounded border pl-2"
        id="transfer"
        name="transfer"
        placeholder="Transfer"
        type="text"
      />
    </div>
    <button
      className="mt-4 bg-purple-500 px-4 py-2 text-white"
      name="transfer"
      type="submit"
    >
      Transfer
    </button>
  </form>,
  <h2>
     
  </h2>,
]
`;
